### ISSUE TITLE
User with a cloned project owner role cannot enable project monitoring

### ISSUE THREAD

**Issue by dnoland1** [rancher/rancher#21355]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
1. Create a cluster, add it to Rancher, create a "projectmon" project.
2. In the global view, go to security -> roles -> projects and select "clone" for the Project Owner role.
3. Name the project "projectownerclone" and change Project Creator Default to no. Keep other default selections. Click Create button.
4. Create a new Standard user called "usermon"
5. Go to the projectownerclone, click Add Member button.
6. Select "usermon" with "projectownerclone" role and click Create.
7. Log in as "usermon" in another browser.
8. Go to the "projectmon" project.
9. Go to Tools -> monitoring
10. Click on the Enable Monitoring button.

**Result:**
Seeing error `EnableMonitoring Error Unknown action: enableMonitoring` and `Error Unknown action: enableMonitoring`

**Other details that may be helpful:**
Expectation is that if project owner role is cloned and applied to a member in the project, that user should have ability to enable monitoring.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
v2.2.4
- Installation option (single install/HA):
single

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported):
Imported

- Machine type (cloud/VM/metal) and specifications (CPU/memory):
k3d
- Kubernetes version (use `kubectl version`):

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.2", GitCommit:"66049e3b21efe110454d67df4fa62b08ea79a19b", GitTreeState:"clean", BuildDate:"2019-05-16T18:55:03Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1-k3s.4", GitCommit:"52f3b42401c93c36467f1fd6d294a3aba26c7def", GitTreeState:"clean", BuildDate:"2019-04-15T22:13+00:00Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
$ docker version
Client: Docker Engine - Community
 Version:           18.09.2
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        6247962
 Built:             Sun Feb 10 04:12:39 2019
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.2
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.6
  Git commit:       6247962
  Built:            Sun Feb 10 04:13:06 2019
  OS/Arch:          linux/amd64
  Experimental:     true
```

**Comment by vincent99** [comment]
Tracking that it was cloned from project owner makes no sense. You could clone it and remove every permission from it and wouldn't expect that role to grant access to anything.

It should be checking for some actual set of RBAC permissions that you have, not for a particular role name (or clone source).

**Comment by dnoland1** [comment]
Agree, but if a project owner seems to have the permission and that role is cloned while carrying over all the permissions, a user in that cloned role should have the permission. Guess the question is what grants ability to enabling monitoring?

**Comment by deniseschannon** [comment]
@loganhz what is the permission/resource that is needed to be able to enable project monitoring?

**Comment by orangedeng** [comment]
We are checking the user-can-update the project which monitoring is going to be enabled. We don't check the specific role name for project in api level.

**Comment by dnoland1** [comment]
@orangedeng I don't think the expectation is to check the role name. The expectation is if you clone the project owner role and assign it to a user without modifying the new role's permissions, that user should be able to enable project monitoring. Does that make sense?

**Comment by vincent99** [comment]
Nobody is suggesting that's what it _should_ do, I was speculating that's what it _seems to be doing_ based on the behavior you're reporting.

But it seems if it actually did what @orangedeng is saying (which _does_ sound like how it _should_ be implemented) then it should already work because the cloned tile world have the same rules.

**Comment by jiaqiluo** [comment]
The bug is reproduced in Rancher:v2.2.6

Steps:
- Run Rancher:v2.2.6 container single install
- In the global view, go to security -> roles -> projects and select "clone" for the Project Owner role.
- Name the project "p-clone" and change Project Creator Default to no. Keep other default selections. Click the Create button.
- Add an RKE DO cluster 
- Created a project "p-1" 
- Enable Github Auth
- add a Github user to the proejct p-1 with the role  p-clone
- log in as the Github user
- go to the project p-1 to enable project monitoring

Results:
<img width="374" alt="Screen Shot 2019-08-06 at 3 24 52 PM" src="https://user-images.githubusercontent.com/6218999/62583164-e5e87280-b863-11e9-92e5-89d5d306f17c.png">

**Comment by jiaqiluo** [comment]
The bug fix is validated in Rancher: master bc3700f8

same steps, now the Github user with the role `p-clone` can enable the cluster monitoring
